File: Data Structures & Algorithms/search-2d-matrix/submission-1.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int start=0;
        int end=(matrix.length*matrix[0].length)-1;
        int columns=matrix[0].length;
        while(start<=end){
            int mid=start+(end-start)/2;
            if(matrix[mid/columns][mid%columns]==target){
                return true;
            }else if(matrix[mid/columns][mid%columns]>target){
                end=mid-1;
            }else{
                start=mid+1;
            }
        }
        return false;
    }
}

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-1.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        start=0
        end=len(nums)-1
        res=-1
        while(start<=end):
            mid=start+(end-start)//2
            if(nums[mid]==target):
                res=mid
                return res
            if(nums[mid]>=nums[start]):
                #left side is sorted
                if(target>=nums[start] and target<nums[mid]):
                    end=mid-1
                else:
                    start=mid+1
            else:
                #right side is sorted
                if(target>nums[mid] and target<=nums[end]):
                    start=mid+1
                else:
                    end=mid-1
        return res
        
        